compiler/src/dotty/tools/dotc/core/TypeEval.scala

```diff
@@ -111,6 +111,15 @@ object TypeEval:
               nestedPairs(fieldLabels) :: nestedPairs(fieldTypes) :: Nil
         else arg.widenDealias match
           case arg @ defn.NamedTuple(_, _) => Some(arg)
+          case arg if arg.derivesFrom(defn.TupleClass) =>
+            val fieldTypesOpt = tupleElementTypes(arg)
+            fieldTypesOpt match
+              case Some(fieldTypes) =>
+                val fieldLabels = fieldTypes.zipWithIndex.map((_, i) => ConstantType(Constant(s"_$i")))
```

Comment:
The first field of a tuple is `_1` and not `_0`. Also to avoid materializing an intermediate collection with zipWithIndex we could do something like:
```suggestion
                val fieldLabels = (for i <- 1 to fieldTypes.length yield ConstantType(Constant(s"_$i")))).toList
```